controller/array_action/array_mediator_xiv.py

```diff
@@ -264,6 +267,25 @@ def create_snapshot(self, name, volume_name):
             raise controller_errors.PermissionDeniedError(
                 "create snapshot {0} from volume {1}".format(name, volume_name))
 
+    def delete_snapshot(self, snapshot_id):
+        logger.info("Deleting snapshot with id : {0}".format(snapshot_id))
+        try:
+            snapshot_name = self._get_vol_by_wwn(snapshot_id)
+        except controller_errors.VolumeNotFoundError:
+            raise controller_errors.SnapshotNotFoundError(snapshot_id)
+
+        try:
+            self.client.cmd.snapshot_delete(snapshot=snapshot_name)
```

Comment:
why does `SnapshotIdBelongsToVolumeError` exist in the interface only for "create volume from snapshot" scenario, and not here? can't we get `CommandFailedRuntimeError("The operation is permitted on snapshots only.")` here?